pkg/controllers/cloud/configmap_controller.go

```diff
@@ -54,43 +82,78 @@ func (r *ConfigMapReconciler) Reconcile(_ context.Context, req ctrl.Request) (ct
 	if len(prefix) == 0 {
-		r.Log.Info("Setting CloudResourcePrefix to default", "Old", r.ControllerConfig.CloudResourcePrefix,
-			"New", config.DefaultCloudResourcePrefix)
+		r.Log.Info("Setting CloudResourcePrefix to default", "CloudResourcePrefix", config.DefaultCloudResourcePrefix)
 		r.ControllerConfig.CloudResourcePrefix = config.DefaultCloudResourcePrefix
 		return nil
 	}
 
 	if !(config.ValidateName(prefix)) {
 		return fmt.Errorf("invalid CloudResourcePrefix %s, "+
-			"First and last characters should be alphanumeric and "+
-			"'-' characters are allowed only in the middle", prefix)
+			"Only alphanumeric and '-' characters are allowed."+
+			" Special character '-' is only allowed at the middle", prefix)
 	}
 
-	// Update the CloudResourcePrefix
+	// Update the CloudResourcePrefix.
 	if r.ControllerConfig.CloudResourcePrefix != prefix {
-		r.Log.V(1).Info("Updating the CloudResourcePrefix",
-			"Old", r.ControllerConfig.CloudResourcePrefix, "New", prefix)
+		r.Log.Info("Updating the CloudResourcePrefix", "CloudResourcePrefix", prefix)
 		r.ControllerConfig.CloudResourcePrefix = prefix
 	}
 	return nil
 }
 
+// setCloudSyncInterval sets the CloudSyncInterval.
+func (r *ConfigMapReconciler) setCloudSyncInterval(interval int64) {
+	if interval == 0 {
+		r.Log.Info("Setting CloudSyncInterval to default", "CloudSyncInterval", config.DefaultCloudSyncInterval)
+		r.ControllerConfig.CloudSyncInterval = config.DefaultCloudSyncInterval
+	} else if interval < config.DefaultCloudSyncInterval {
+		r.ControllerConfig.CloudSyncInterval = config.DefaultCloudSyncInterval
+		r.Log.Info(fmt.Sprintf("Invalid CloudSyncInterval %d"+
+			"CloudSyncInterval should be >= %d seconds. So using default interval", interval, config.DefaultCloudSyncInterval))
+	} else if interval != r.ControllerConfig.CloudSyncInterval {
+		// Update the CloudSyncInterval.
+		r.Log.Info("Updating the CloudSyncInterval", "CloudSyncInterval", interval)
+		r.ControllerConfig.CloudSyncInterval = interval
+	} else {
+		return
+	}
+	r.notifyConfigMapHandlers(cloudSyncIntervalConfig)
+}
+
+func (r *ConfigMapReconciler) registerConfigMapHandlers(handler ConfigMapHandler, configType controllerConfigType) {
+	r.configmapHandlerFunc[configType] = append(r.configmapHandlerFunc[configType], handler)
+}
+
+func (r *ConfigMapReconciler) notifyConfigMapHandlers(configType controllerConfigType) {
+	for i := 0; i < len(r.configmapHandlerFunc[configType]); i++ {
+		r.configmapHandlerFunc[configType][i](r.ControllerConfig)
+	}
+}
+
 // SetupWithManager registers ConfigMapReconciler with manager.
 func (r *ConfigMapReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	r.ControllerConfig = &config.ControllerConfig{
 		CloudResourcePrefix: config.DefaultCloudResourcePrefix,
+		CloudSyncInterval:   config.DefaultCloudSyncInterval,
 	}
+	r.configmapHandlerFunc = make(map[controllerConfigType][]ConfigMapHandler)
+	setConfigMapControllerInstance(r)
```

Comment:
This will not be needed when you move to init

Reply:
We can initialize this on this method right.

Reply:
Solved.